### PR TITLE
[labeling] fix BR in HTML labels

### LINKED ISSUE #55532: Linebreak and text vanish in HTML label
### What is the bug or the crash?

I tried to create some fancy labels for a vector layer with HTML formatting but some parts are not displayed, parts of my text are vanishing.

On the left is my label expression editor open, you can see the intended rendering in the preview.  
On the right is one such label in a map canvas.

image

### Steps to reproduce the issue

1. Use this as a label for a layer: `foo<br>bar<span style="color:red">baz</span>`
2. Enable HTML rendering for the label
3. Look at such label in the map, it reads "foo     baz" with "baz" shown in red, "bar" is not shown although space seems reserved for it

### Versions

<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.0//EN" "http://www.w3.org/TR/REC-html40/strict.dtd">
<html><head><meta http-equiv="Content-Type" content="text/html; charset=utf-8" /><style type="text/css">
p, li { white-space: pre-wrap; }
</style></head><body>

QGIS-Version | 3.34.0-Prizren | QGIS-Codezweig | Release 3.34
-- | -- | -- | --
Qt-Version | 5.15.11
Python-Version | 3.11.5
GDAL-Version | 3.7.3
PROJ-Version | 9.3.0
EPSG-Registraturdatenbankversion | v10.094 (2023-08-08)
GEOS-Version | 3.12.0-CAPI-1.18.0
Kompiliert mit SQLite | 3.44.0 | Läuft mit SQLite | 3.44.2
Gegen PDAL kompiliert | 2.6.0 | Läuft mit PDAL | 2.6.1
PostgreSQL-Client-Version | 16.0
SpatiaLite-Version | 5.1.0
QWT-Version | 6.2.0
QScintilla2-Version | 2.14.1
BS-Version | Manjaro Linux
  |   |   |  
Aktive Python-Erweiterungen
searchlayers | 3.0.14
MetaSearch | 0.3.6
db_manager | 0.1.20
grassprovider | 2.12.99
processing | 2.12.99

</body></html>QGIS-Version
3.34.0-Prizren
QGIS-Codezweig
[Release 3.34](https://github.com/qgis/QGIS/tree/release-3_34)
Qt-Version
5.15.11
Python-Version
3.11.5
GDAL-Version
3.7.3
PROJ-Version
9.3.0
EPSG-Registraturdatenbankversion
v10.094 (2023-08-08)
GEOS-Version
3.12.0-CAPI-1.18.0
Kompiliert mit SQLite
3.44.0
Läuft mit SQLite
3.44.2
Gegen PDAL kompiliert
2.6.0
Läuft mit PDAL
2.6.1
PostgreSQL-Client-Version
16.0
SpatiaLite-Version
5.1.0
QWT-Version
6.2.0
QScintilla2-Version
2.14.1
BS-Version
Manjaro Linux

Aktive Python-Erweiterungen
searchlayers
3.0.14
MetaSearch
0.3.6
db_manager
0.1.20
grassprovider
2.12.99
processing
2.12.99

### Supported QGIS version

- [X] I'm running a supported QGIS version according to [the roadmap](https://www.qgis.org/en/site/getinvolved/development/roadmap.html#release-schedule).

### New profile

- [X] I tried with a new [QGIS profile](https://docs.qgis.org/latest/en/docs/user_manual/introduction/qgis_configuration.html#working-with-user-profiles)

### Additional context

If I use `<p></p>` instead of a `<br>` or `<br />`, it looks as intended.

### PR DESCRIPTION
**Pull request by elpaso**
Fix `<br>` ignored in HTML formatted labels.

Fix #55532

### CODE REVIEW

**Comment by elpaso** [comment]
CC @nyalldawson

File: src/core/textrenderer/qgstextdocument.cpp [hunk 1]
```diff
@@ -69,18 +74,67 @@ QgsTextDocument QgsTextDocument::fromHtml( const QStringList &lines )
...
+              splitLineBlock.append( splitFragment );
+              document.append( splitLineBlock );
+
+            }
+          }
+          else
+          {
+            block.append( QgsTextFragment( fragment ) );
+          }
+
```
Comment (elpaso) [inline src/core/textrenderer/qgstextdocument.cpp:123]:
```suggestion
```

**Review by nirvn — APPROVED** [review]
Logic looks good to me.

File: src/core/textrenderer/qgstextdocument.cpp [hunk 2]
```diff
@@ -69,18 +74,67 @@ QgsTextDocument QgsTextDocument::fromHtml( const QStringList &lines )
...
       }
+
       if ( !block.empty() )
         document.append( block );
 
       sourceBlock = sourceBlock.next();
       if ( !sourceBlock.isValid() )
         break;
     }
+
```
Comment (elpaso) [inline src/core/textrenderer/qgstextdocument.cpp:135]:
```suggestion
```

**Pull request merged**